// @azure/arm-containerservice/dist/react-native/models/parameters.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { ManagedCluster as ManagedClusterMapper, TagsObject as TagsObjectMapper, ManagedClusterServicePrincipalProfile as ManagedClusterServicePrincipalProfileMapper, ManagedClusterAADProfile as ManagedClusterAADProfileMapper, RunCommandRequest as RunCommandRequestMapper, MaintenanceConfiguration as MaintenanceConfigurationMapper, ManagedNamespace as ManagedNamespaceMapper, AgentPool as AgentPoolMapper, AgentPoolDeleteMachinesParameter as AgentPoolDeleteMachinesParameterMapper, PrivateEndpointConnection as PrivateEndpointConnectionMapper, PrivateLinkResource as PrivateLinkResourceMapper, Snapshot as SnapshotMapper, TrustedAccessRoleBinding as TrustedAccessRoleBindingMapper, } from "../models/mappers.js";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2025-10-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String",
        },
    },
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "Uuid",
        },
    },
};
export const location = {
    parameterPath: "location",
    mapper: {
        constraints: {
            MinLength: 1,
        },
        serializedName: "location",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1,
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const resourceName = {
    parameterPath: "resourceName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-zA-Z0-9]$|^[a-zA-Z0-9][-_a-zA-Z0-9]{0,61}[a-zA-Z0-9]$"),
            MaxLength: 63,
            MinLength: 1,
        },
        serializedName: "resourceName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const roleName = {
    parameterPath: "roleName",
    mapper: {
        serializedName: "roleName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const serverFqdn = {
    parameterPath: ["options", "serverFqdn"],
    mapper: {
        serializedName: "server-fqdn",
        type: {
            name: "String",
        },
    },
};
export const format = {
    parameterPath: ["options", "format"],
    mapper: {
        serializedName: "format",
        type: {
            name: "String",
        },
    },
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
export const parameters = {
    parameterPath: "parameters",
    mapper: ManagedClusterMapper,
};
export const ifMatch = {
    parameterPath: ["options", "ifMatch"],
    mapper: {
        serializedName: "If-Match",
        type: {
            name: "String",
        },
    },
};
export const ifNoneMatch = {
    parameterPath: ["options", "ifNoneMatch"],
    mapper: {
        serializedName: "If-None-Match",
        type: {
            name: "String",
        },
    },
};
export const parameters1 = {
    parameterPath: "parameters",
    mapper: TagsObjectMapper,
};
export const parameters2 = {
    parameterPath: "parameters",
    mapper: ManagedClusterServicePrincipalProfileMapper,
};
export const parameters3 = {
    parameterPath: "parameters",
    mapper: ManagedClusterAADProfileMapper,
};
export const requestPayload = {
    parameterPath: "requestPayload",
    mapper: RunCommandRequestMapper,
};
export const commandId = {
    parameterPath: "commandId",
    mapper: {
        serializedName: "commandId",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const mode = {
    parameterPath: "mode",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-zA-Z0-9]$|^[a-zA-Z0-9][-_a-zA-Z0-9]{0,61}[a-zA-Z0-9]$"),
            MaxLength: 24,
            MinLength: 1,
        },
        serializedName: "mode",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
export const configName = {
    parameterPath: "configName",
    mapper: {
        serializedName: "configName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const parameters4 = {
    parameterPath: "parameters",
    mapper: MaintenanceConfigurationMapper,
};
export const managedNamespaceName = {
    parameterPath: "managedNamespaceName",
    mapper: {
        constraints: {
            Pattern: new RegExp("[a-z0-9]([-a-z0-9]*[a-z0-9])?"),
            MaxLength: 63,
            MinLength: 1,
        },
        serializedName: "managedNamespaceName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const parameters5 = {
    parameterPath: "parameters",
    mapper: ManagedNamespaceMapper,
};
export const agentPoolName = {
    parameterPath: "agentPoolName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-z][a-z0-9]{0,11}$"),
            MaxLength: 12,
            MinLength: 1,
        },
        serializedName: "agentPoolName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const parameters6 = {
    parameterPath: "parameters",
    mapper: AgentPoolMapper,
};
export const ignorePodDisruptionBudget = {
    parameterPath: ["options", "ignorePodDisruptionBudget"],
    mapper: {
        serializedName: "ignore-pod-disruption-budget",
        type: {
            name: "Boolean",
        },
    },
};
export const machines = {
    parameterPath: "machines",
    mapper: AgentPoolDeleteMachinesParameterMapper,
};
export const privateEndpointConnectionName = {
    parameterPath: "privateEndpointConnectionName",
    mapper: {
        serializedName: "privateEndpointConnectionName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const parameters7 = {
    parameterPath: "parameters",
    mapper: PrivateEndpointConnectionMapper,
};
export const parameters8 = {
    parameterPath: "parameters",
    mapper: PrivateLinkResourceMapper,
};
export const parameters9 = {
    parameterPath: "parameters",
    mapper: SnapshotMapper,
};
export const trustedAccessRoleBindingName = {
    parameterPath: "trustedAccessRoleBindingName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^([A-Za-z0-9-])+$"),
            MaxLength: 24,
            MinLength: 1,
        },
        serializedName: "trustedAccessRoleBindingName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const trustedAccessRoleBinding = {
    parameterPath: "trustedAccessRoleBinding",
    mapper: TrustedAccessRoleBindingMapper,
};
export const machineName = {
    parameterPath: "machineName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-zA-Z0-9][-_a-zA-Z0-9]{0,39}$"),
        },
        serializedName: "machineName",
        required: true,
        type: {
            name: "String",
        },
    },
};
